core/Plugin/Manager.php

```diff
@@ -838,37 +835,56 @@ private function reloadActivatedPlugins()
     {
         $pluginsToPostPendingEventsTo = array();
         foreach ($this->pluginsToLoad as $pluginName) {
-            if (!$this->isPluginLoaded($pluginName)
-                && !$this->isPluginThirdPartyAndBogus($pluginName)
-            ) {
-                $newPlugin = $this->loadPlugin($pluginName);
-                if ($newPlugin === null) {
-                    continue;
-                }
+            $pluginsToPostPendingEventsTo = $this->reloadActivatedPlugin($pluginName, $pluginsToPostPendingEventsTo);
+        }
 
-                if ($newPlugin->hasMissingDependencies()) {
-                    $this->deactivatePlugin($pluginName);
+        // post pending events after all plugins are successfully loaded
+        foreach ($pluginsToPostPendingEventsTo as $plugin) {
+            EventDispatcher::getInstance()->postPendingEventsTo($plugin);
+        }
+    }
 
-                    // at this state we do not know yet whether current user has super user access. We do not even know
-                    // if someone is actually logged in.
-                    $message  = Piwik::translate('CorePluginsAdmin_WeDeactivatedThePluginAsItHasMissingDependencies', array($pluginName, $newPlugin->getMissingDependenciesAsString()));
-                    $message .= ' ';
-                    $message .= Piwik::translate('General_PleaseContactYourPiwikAdministrator');
+    private function reloadActivatedPlugin($pluginName, $pluginsToPostPendingEventsTo)
+    {
+        if ($this->isPluginLoaded($pluginName) || $this->isPluginThirdPartyAndBogus($pluginName)) {
+            return $pluginsToPostPendingEventsTo;
+        }
 
-                    $notification = new Notification($message);
-                    $notification->context = Notification::CONTEXT_ERROR;
-                    Notification\Manager::notify('PluginManager_PluginDeactivated', $notification);
-                    continue;
-                }
+        $newPlugin = $this->loadPlugin($pluginName);
 
-                $pluginsToPostPendingEventsTo[] = $newPlugin;
+        if ($newPlugin === null) {
+            return $pluginsToPostPendingEventsTo;
+        }
+
+        $requirements = $newPlugin->getMissingDependencies();
+
+        if (!empty($requirements)) {
```

Comment:
Yes, that should still work the same. The logic should be the same, we only load another plugin first under circumstances. I just also debugged and events are postponed to all plugins and if there is a missing dependency it will postpone events to a dependency before the plugin that requires this dependency. Which is good as then a plugin that requires a different plugin could overwrite an action etc.